packages/react-components/react-spinner/src/contexts/SpinnerContext.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import type { SpinnerSize } from '../components/Spinner/Spinner.types';
+
+const SpinnerContext = React.createContext<SpinnerContextValue | undefined>(undefined);
+
+/**
+ * @internal
+ */
+export interface SpinnerContextValue {
+  size?: SpinnerSize;
+}
+
+const SpinnerContextDefaultValue: SpinnerContextValue = { size: 'medium' };
+
+/**
+ * @internal
+ */
+export const SpinnerContextProvider = SpinnerContext.Provider;
+
+/**
+ * @internal
+ */
+export const useSpinnerContext = () => React.useContext(SpinnerContext) ?? SpinnerContextDefaultValue;
```

Comment:
Any reason not to use react-context-selector? https://github.com/microsoft/fluentui/tree/master/packages/react-components/react-shared-contexts

Here's an example of the use of it: https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-avatar/src/contexts/AvatarGroupContext.ts

Reply:
The main reason to use context selectors are when the context is not a primitive (i.e. object) and when the properties change very often (i.e. open value that is toggled), which will avoid excessive re-renders when using context based state.

In this case, is there an expectation that values in this context would change often?

Reply:
Values in this context would not change very often, but the context is an enum for the size prop